{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " Cutadapt:\n",
    " \n",
    " http://cutadapt.readthedocs.io/en/stable/guide.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import xml.etree.ElementTree as ET\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "untrimmed_seqs_path = '../../Yeast2HybridData/NextSeq-2022-01-10-L68/Untrimmed/'\n",
    "trimmed_seqs_path = '../../Yeast2HybridData/NextSeq-2022-01-10-L68/Trimmed/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "if not os.path.exists(untrimmed_seqs_path):\n",
    "    os.makedirs(untrimmed_seqs_path)\n",
    "    os.makedirs(trimmed_seqs_path)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Remove adaptors"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### list of fastq filenames"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#files = [filename for filename in os.listdir(untrimmed_seqs_path) if filename[-6:] == '.fastq']\n",
    "files = [filename for filename in os.listdir(untrimmed_seqs_path) if filename[-3:] == '.gz']\n",
    "files = sorted(files)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['L68_HIS_S2_R1_001.fastq.gz', 'L68_TRP_S1_R1_001.fastq.gz']"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Define sequences surrounding the binders and the barcodes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# This version is for 71/237bp read1/read2\n",
    "\n",
    "barcode1_prefix = \"CGGGGCGGGGTTATTACGACAT\"\n",
    "barcode1_suffix = \"CCTCCAGACAGCGTTCCGTTA\"\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Indexes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "files_dict = {}\n",
    "\n",
    "for f in files:\n",
    "    \n",
    "    files_dict[f] = [barcode1_prefix, barcode1_suffix]\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'L68_HIS_S2_R1_001.fastq.gz': ['CGGGGCGGGGTTATTACGACAT',\n",
       "  'CCTCCAGACAGCGTTCCGTTA'],\n",
       " 'L68_TRP_S1_R1_001.fastq.gz': ['CGGGGCGGGGTTATTACGACAT',\n",
       "  'CCTCCAGACAGCGTTCCGTTA']}"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files_dict"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### loop through and trim each fastq file with cutadapt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 1.18 with Python 3.6.9\n",
      "Command line parameters: -g CGGGGCGGGGTTATTACGACAT...CCTCCAGACAGCGTTCCGTTA -m 17 -M 23 -e 0.1 -o ../../Yeast2HybridData/NextSeq-2022-01-10-L68/Trimmed/L68_HIS_S2_R1_001.fastq.gz ../../Yeast2HybridData/NextSeq-2022-01-10-L68/Untrimmed/L68_HIS_S2_R1_001.fastq.gz\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 213.92 s (26 us/read; 2.35 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               8,375,754\n",
      "Reads with adapters:                 8,339,898 (99.6%)\n",
      "Reads that were too short:               8,966 (0.1%)\n",
      "Reads that were too long:               44,765 (0.5%)\n",
      "Reads written (passing filters):     8,322,023 (99.4%)\n",
      "\n",
      "Total basepairs processed: 2,052,059,730 bp\n",
      "Total written (filtered):    166,431,486 bp (8.1%)\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CGGGGCGGGGTTATTACGACAT...CCTCCAGACAGCGTTCCGTTA; Type: linked; Length: 22+21; 5' trimmed: 8339898 times; 3' trimmed: 8339898 times\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-21 bp: 2\n",
      "\n",
      "Overview of removed sequences at 5' end\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t146\t130871.2\t0\t146\n",
      "4\t60\t32717.8\t0\t60\n",
      "5\t10\t8179.4\t0\t10\n",
      "6\t1\t2044.9\t0\t1\n",
      "7\t2\t511.2\t0\t2\n",
      "22\t1\t0.0\t2\t0 1\n",
      "43\t1\t0.0\t2\t1\n",
      "51\t1\t0.0\t2\t0 0 1\n",
      "52\t3\t0.0\t2\t3\n",
      "53\t1\t0.0\t2\t1\n",
      "55\t1\t0.0\t2\t1\n",
      "56\t3\t0.0\t2\t2 0 1\n",
      "57\t19\t0.0\t2\t17 2\n",
      "58\t12\t0.0\t2\t11 0 1\n",
      "59\t17\t0.0\t2\t17\n",
      "60\t27\t0.0\t2\t23 1 3\n",
      "61\t57\t0.0\t2\t54 2 1\n",
      "62\t61\t0.0\t2\t57 2 2\n",
      "63\t122\t0.0\t2\t114 4 4\n",
      "64\t105\t0.0\t2\t95 4 6\n",
      "65\t141\t0.0\t2\t132 3 6\n",
      "66\t196\t0.0\t2\t162 4 30\n",
      "67\t783\t0.0\t2\t218 7 558\n",
      "68\t70524\t0.0\t2\t323 56 70145\n",
      "69\t868\t0.0\t2\t668 24 176\n",
      "70\t1096\t0.0\t2\t1047 34 15\n",
      "71\t3858\t0.0\t2\t3484 162 212\n",
      "72\t82446\t0.0\t2\t68067 11482 2897\n",
      "73\t8149144\t0.0\t2\t8010398 126241 12505\n",
      "74\t26855\t0.0\t2\t19307 6879 669\n",
      "75\t297\t0.0\t2\t186 32 79\n",
      "76\t79\t0.0\t2\t78 0 1\n",
      "77\t80\t0.0\t2\t76 2 2\n",
      "78\t2679\t0.0\t2\t2633 39 7\n",
      "79\t28\t0.0\t2\t28\n",
      "80\t30\t0.0\t2\t30\n",
      "81\t25\t0.0\t2\t25\n",
      "82\t8\t0.0\t2\t8\n",
      "83\t22\t0.0\t2\t22\n",
      "84\t5\t0.0\t2\t5\n",
      "85\t7\t0.0\t2\t7\n",
      "86\t7\t0.0\t2\t7\n",
      "87\t8\t0.0\t2\t8\n",
      "88\t4\t0.0\t2\t3 0 1\n",
      "89\t3\t0.0\t2\t3\n",
      "90\t7\t0.0\t2\t6 1\n",
      "91\t2\t0.0\t2\t2\n",
      "92\t6\t0.0\t2\t4 0 2\n",
      "93\t3\t0.0\t2\t3\n",
      "94\t4\t0.0\t2\t4\n",
      "95\t3\t0.0\t2\t3\n",
      "96\t4\t0.0\t2\t4\n",
      "97\t4\t0.0\t2\t3 0 1\n",
      "98\t3\t0.0\t2\t3\n",
      "99\t3\t0.0\t2\t3\n",
      "100\t1\t0.0\t2\t1\n",
      "101\t4\t0.0\t2\t3 1\n",
      "102\t3\t0.0\t2\t3\n",
      "107\t1\t0.0\t2\t1\n",
      "109\t3\t0.0\t2\t3\n",
      "110\t1\t0.0\t2\t1\n",
      "112\t1\t0.0\t2\t1\n",
      "114\t1\t0.0\t2\t1\n",
      "123\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "Overview of removed sequences at 3' end\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1\t130871.2\t0\t1\n",
      "69\t28\t0.0\t2\t25 2 1\n",
      "72\t4\t0.0\t2\t0 2 2\n",
      "73\t11\t0.0\t2\t5 4 2\n",
      "74\t3\t0.0\t2\t2 1\n",
      "96\t1\t0.0\t2\t1\n",
      "98\t12\t0.0\t2\t11 1\n",
      "102\t1\t0.0\t2\t1\n",
      "111\t1\t0.0\t2\t1\n",
      "113\t1\t0.0\t2\t1\n",
      "115\t1\t0.0\t2\t1\n",
      "116\t3\t0.0\t2\t3\n",
      "118\t2\t0.0\t2\t2\n",
      "123\t3\t0.0\t2\t1 2\n",
      "124\t4\t0.0\t2\t4\n",
      "125\t1\t0.0\t2\t1\n",
      "126\t3\t0.0\t2\t3\n",
      "127\t3\t0.0\t2\t3\n",
      "128\t4\t0.0\t2\t4\n",
      "129\t4\t0.0\t2\t4\n",
      "130\t42\t0.0\t2\t40 2\n",
      "131\t6\t0.0\t2\t5 1\n",
      "132\t23\t0.0\t2\t22 1\n",
      "133\t8420\t0.0\t2\t8018 317 85\n",
      "134\t78\t0.0\t2\t68 10\n",
      "135\t11\t0.0\t2\t9 2\n",
      "136\t12\t0.0\t2\t10 2\n",
      "137\t14\t0.0\t2\t12 2\n",
      "138\t36\t0.0\t2\t26 7 3\n",
      "139\t9\t0.0\t2\t6 2 1\n",
      "140\t18\t0.0\t2\t7 1 10\n",
      "141\t9\t0.0\t2\t6 0 3\n",
      "142\t23\t0.0\t2\t22 1\n",
      "143\t9\t0.0\t2\t8 0 1\n",
      "144\t29\t0.0\t2\t23 3 3\n",
      "145\t33\t0.0\t2\t27 4 2\n",
      "146\t30\t0.0\t2\t30\n",
      "147\t2680\t0.0\t2\t2582 76 22\n",
      "148\t79\t0.0\t2\t75 3 1\n",
      "149\t84\t0.0\t2\t82 1 1\n",
      "150\t216\t0.0\t2\t205 10 1\n",
      "151\t25827\t0.0\t2\t24604 1017 206\n",
      "152\t8125542\t0.0\t2\t7873223 220562 31757\n",
      "153\t89458\t0.0\t2\t81694 6987 777\n",
      "154\t3954\t0.0\t2\t3773 154 27\n",
      "155\t1258\t0.0\t2\t1149 93 16\n",
      "156\t9616\t0.0\t2\t9167 367 82\n",
      "157\t70629\t0.0\t2\t67546 2710 373\n",
      "158\t818\t0.0\t2\t781 29 8\n",
      "159\t197\t0.0\t2\t181 11 5\n",
      "160\t145\t0.0\t2\t136 7 2\n",
      "161\t105\t0.0\t2\t99 6\n",
      "162\t126\t0.0\t2\t115 9 2\n",
      "163\t62\t0.0\t2\t56 4 2\n",
      "164\t117\t0.0\t2\t110 5 2\n",
      "165\t25\t0.0\t2\t24 1\n",
      "166\t17\t0.0\t2\t17\n",
      "167\t11\t0.0\t2\t11\n",
      "168\t28\t0.0\t2\t24 2 2\n",
      "169\t3\t0.0\t2\t3\n",
      "170\t1\t0.0\t2\t0 1\n",
      "172\t1\t0.0\t2\t1\n",
      "173\t3\t0.0\t2\t3\n",
      "174\t1\t0.0\t2\t1\n",
      "182\t1\t0.0\t2\t1\n",
      "203\t1\t0.0\t2\t1\n",
      "\n",
      "This is cutadapt 1.18 with Python 3.6.9\n",
      "Command line parameters: -g CGGGGCGGGGTTATTACGACAT...CCTCCAGACAGCGTTCCGTTA -m 17 -M 23 -e 0.1 -o ../../Yeast2HybridData/NextSeq-2022-01-10-L68/Trimmed/L68_TRP_S1_R1_001.fastq.gz ../../Yeast2HybridData/NextSeq-2022-01-10-L68/Untrimmed/L68_TRP_S1_R1_001.fastq.gz\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 203.38 s (26 us/read; 2.32 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               7,872,442\n",
      "Reads with adapters:                 7,832,446 (99.5%)\n",
      "Reads that were too short:               1,824 (0.0%)\n",
      "Reads that were too long:               43,597 (0.6%)\n",
      "Reads written (passing filters):     7,827,021 (99.4%)\n",
      "\n",
      "Total basepairs processed: 1,928,748,290 bp\n",
      "Total written (filtered):    156,534,492 bp (8.1%)\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CGGGGCGGGGTTATTACGACAT...CCTCCAGACAGCGTTCCGTTA; Type: linked; Length: 22+21; 5' trimmed: 7832446 times; 3' trimmed: 7832446 times\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-21 bp: 2\n",
      "\n",
      "Overview of removed sequences at 5' end\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t133\t123006.9\t0\t133\n",
      "4\t47\t30751.7\t0\t47\n",
      "5\t9\t7687.9\t0\t9\n",
      "6\t3\t1922.0\t0\t3\n",
      "10\t1\t7.5\t1\t0 1\n",
      "51\t1\t0.0\t2\t1\n",
      "52\t2\t0.0\t2\t2\n",
      "54\t3\t0.0\t2\t3\n",
      "55\t66\t0.0\t2\t66\n",
      "56\t7\t0.0\t2\t6 1\n",
      "57\t18\t0.0\t2\t17 0 1\n",
      "58\t15\t0.0\t2\t15\n",
      "59\t32\t0.0\t2\t32\n",
      "60\t21\t0.0\t2\t17 4\n",
      "61\t401\t0.0\t2\t314 86 1\n",
      "62\t52\t0.0\t2\t49 3\n",
      "63\t280\t0.0\t2\t269 10 1\n",
      "64\t83\t0.0\t2\t81 2\n",
      "65\t105\t0.0\t2\t101 3 1\n",
      "66\t323\t0.0\t2\t309 11 3\n",
      "67\t244\t0.0\t2\t224 10 10\n",
      "68\t1332\t0.0\t2\t317 16 999\n",
      "69\t852\t0.0\t2\t829 17 6\n",
      "70\t1356\t0.0\t2\t1305 37 14\n",
      "71\t4708\t0.0\t2\t3836 236 636\n",
      "72\t87144\t0.0\t2\t67977 16236 2931\n",
      "73\t7702863\t0.0\t2\t7571762 117917 13184\n",
      "74\t30587\t0.0\t2\t23894 5779 914\n",
      "75\t854\t0.0\t2\t297 17 540\n",
      "76\t87\t0.0\t2\t76 6 5\n",
      "77\t49\t0.0\t2\t48 1\n",
      "78\t278\t0.0\t2\t271 6 1\n",
      "79\t12\t0.0\t2\t12\n",
      "80\t10\t0.0\t2\t10\n",
      "81\t10\t0.0\t2\t10\n",
      "82\t20\t0.0\t2\t18 1 1\n",
      "83\t341\t0.0\t2\t329 11 1\n",
      "84\t9\t0.0\t2\t9\n",
      "85\t3\t0.0\t2\t3\n",
      "86\t5\t0.0\t2\t5\n",
      "87\t5\t0.0\t2\t3 0 2\n",
      "88\t5\t0.0\t2\t5\n",
      "89\t2\t0.0\t2\t2\n",
      "90\t2\t0.0\t2\t2\n",
      "91\t2\t0.0\t2\t2\n",
      "92\t7\t0.0\t2\t6 0 1\n",
      "93\t2\t0.0\t2\t2\n",
      "94\t4\t0.0\t2\t4\n",
      "95\t5\t0.0\t2\t5\n",
      "97\t1\t0.0\t2\t1\n",
      "98\t1\t0.0\t2\t1\n",
      "99\t3\t0.0\t2\t3\n",
      "100\t2\t0.0\t2\t2\n",
      "101\t1\t0.0\t2\t1\n",
      "102\t3\t0.0\t2\t3\n",
      "103\t4\t0.0\t2\t4\n",
      "104\t6\t0.0\t2\t6\n",
      "107\t3\t0.0\t2\t3\n",
      "109\t1\t0.0\t2\t1\n",
      "118\t1\t0.0\t2\t1\n",
      "121\t1\t0.0\t2\t1\n",
      "122\t1\t0.0\t2\t1\n",
      "123\t1\t0.0\t2\t1\n",
      "151\t2\t0.0\t2\t2\n",
      "152\t5\t0.0\t2\t5\n",
      "156\t8\t0.0\t2\t8\n",
      "179\t2\t0.0\t2\t2\n",
      "\n",
      "\n",
      "Overview of removed sequences at 3' end\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t4\t123006.9\t0\t4\n",
      "4\t2\t30751.7\t0\t2\n",
      "46\t7\t0.0\t2\t2 4 1\n",
      "68\t2\t0.0\t2\t2\n",
      "69\t534\t0.0\t2\t431 74 29\n",
      "70\t2\t0.0\t2\t1 1\n",
      "72\t28\t0.0\t2\t18 5 5\n",
      "73\t162\t0.0\t2\t100 38 24\n",
      "74\t47\t0.0\t2\t35 6 6\n",
      "93\t42\t0.0\t2\t35 5 2\n",
      "94\t1\t0.0\t2\t1\n",
      "97\t3\t0.0\t2\t3\n",
      "98\t71\t0.0\t2\t66 5\n",
      "99\t1\t0.0\t2\t1\n",
      "102\t1\t0.0\t2\t1\n",
      "103\t1\t0.0\t2\t1\n",
      "104\t1\t0.0\t2\t1\n",
      "107\t1\t0.0\t2\t1\n",
      "116\t1\t0.0\t2\t1\n",
      "118\t3\t0.0\t2\t2 1\n",
      "121\t6\t0.0\t2\t5 1\n",
      "122\t4\t0.0\t2\t4\n",
      "123\t3\t0.0\t2\t3\n",
      "124\t1\t0.0\t2\t1\n",
      "125\t2\t0.0\t2\t2\n",
      "126\t3\t0.0\t2\t3\n",
      "127\t1\t0.0\t2\t1\n",
      "128\t1\t0.0\t2\t1\n",
      "130\t15\t0.0\t2\t14 0 1\n",
      "131\t8\t0.0\t2\t8\n",
      "132\t11\t0.0\t2\t10 0 1\n",
      "133\t1977\t0.0\t2\t1874 87 16\n",
      "134\t25\t0.0\t2\t21 4\n",
      "135\t6\t0.0\t2\t3 3\n",
      "136\t14\t0.0\t2\t9 5\n",
      "137\t18\t0.0\t2\t16 2\n",
      "138\t469\t0.0\t2\t437 22 10\n",
      "139\t8\t0.0\t2\t8\n",
      "140\t8\t0.0\t2\t3 1 4\n",
      "141\t11\t0.0\t2\t8 3\n",
      "142\t342\t0.0\t2\t324 14 4\n",
      "143\t20\t0.0\t2\t17 3\n",
      "144\t12\t0.0\t2\t9 2 1\n",
      "145\t11\t0.0\t2\t10 0 1\n",
      "146\t12\t0.0\t2\t9 3\n",
      "147\t278\t0.0\t2\t265 13\n",
      "148\t49\t0.0\t2\t47 2\n",
      "149\t111\t0.0\t2\t102 7 2\n",
      "150\t330\t0.0\t2\t307 20 3\n",
      "151\t31479\t0.0\t2\t29831 1364 284\n",
      "152\t7694767\t0.0\t2\t7436510 224336 33921\n",
      "153\t88657\t0.0\t2\t84160 3753 744\n",
      "154\t5292\t0.0\t2\t5026 220 46\n",
      "155\t1914\t0.0\t2\t1777 103 34\n",
      "156\t1173\t0.0\t2\t1107 55 11\n",
      "157\t1692\t0.0\t2\t1592 86 14\n",
      "158\t720\t0.0\t2\t668 32 20\n",
      "159\t341\t0.0\t2\t317 12 12\n",
      "160\t106\t0.0\t2\t99 5 2\n",
      "161\t83\t0.0\t2\t79 4\n",
      "162\t326\t0.0\t2\t313 12 1\n",
      "163\t52\t0.0\t2\t52\n",
      "164\t840\t0.0\t2\t805 28 7\n",
      "165\t22\t0.0\t2\t20 1 1\n",
      "166\t32\t0.0\t2\t31 1\n",
      "167\t15\t0.0\t2\t15\n",
      "168\t183\t0.0\t2\t175 8\n",
      "169\t8\t0.0\t2\t8\n",
      "170\t68\t0.0\t2\t65 3\n",
      "171\t3\t0.0\t2\t3\n",
      "173\t2\t0.0\t2\t2\n",
      "174\t1\t0.0\t2\t1\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for path, affix in files_dict.items():\n",
    "    # input and output paths\n",
    "    untrimmed = untrimmed_seqs_path + path\n",
    "    trimmed = trimmed_seqs_path + path\n",
    "    \n",
    "    \n",
    "    # just trim off the 5' end\n",
    "    if len(affix) == 1:\n",
    "        suffix = affix[0]\n",
    "        #!~/.local/bin/cutadapt -g {suffix} -m 17 -M 23 -e 0.1 -o {trimmed} {untrimmed}\n",
    "        !cutadapt -g {suffix} -m 7 -M 23 -e 0.1 -o {trimmed} {untrimmed}\n",
    "    \n",
    "    # trim off before and after the barcode\n",
    "    else:\n",
    "        prefix = affix[0]\n",
    "        suffix = affix[1]\n",
    "        #!~/.local/bin/cutadapt -g {prefix}...{suffix} -m 17 -M 23 -e 0.1 -o {trimmed} {untrimmed}\n",
    "        !cutadapt -g {prefix}...{suffix} -m 17 -M 23 -e 0.1 -o {trimmed} {untrimmed}\n",
    "        \n",
    "# -g marks linked adaptors, neither of which are 'anchored'\n",
    "# -m 5 -- this parameter tells cutadapt to throw away reads that have yield a sequence <5 nucleotides long.\n",
    "# If you don't do this, you sometimes end up with empty rows, which fucks up later processing using pandas -- pandas'\n",
    "# read_csv function automatically ignores empty rows, messing up the spacing (the names are kept) and making\n",
    "# life difficult."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
